app/controllers/pxe_controller/iso_datastores.rb

```diff
@@ -114,7 +114,7 @@ def iso_datastore_get_form_vars
   # Set form variables for edit
   def iso_datastore_set_form_vars
     @edit = {}
-    @edit[:emses] = ExtManagementSystem.supporting(:create_iso_datastore).order(:name).select { |ems| ems.supports?(:create_iso_datastore) }
+    @edit[:emses] = ExtManagementSystem.supporting(:create_iso_datastore).order(:name)
```

Comment:
If you flip this, you get the ordering via the database itself as opposed to in Ruby

```suggestion
    @edit[:emses] = ExtManagementSystem.order(:name).supporting(:create_iso_datastore)
```